packages/synapse-react-client/src/components/FullTextSearch/FullTextSearch.tsx

```diff
@@ -0,0 +1,108 @@
+import { Collapse, TextField } from '@mui/material'
+import React, { ChangeEvent, useRef, useState } from 'react'
+import { useQueryContext } from '../QueryContext'
+import { useQueryVisualizationContext } from '../QueryVisualizationWrapper'
+import { HelpPopover } from '../HelpPopover/HelpPopover'
+import IconSvg from '../IconSvg/IconSvg'
+import { IconSvgButton } from '../IconSvgButton'
+import { useQuery } from '@tanstack/react-query'
+import { updateQueryUsingSearchTerm } from './FullTextSearchUtils'
+
+// See PLFM-7011
+const MIN_SEARCH_QUERY_LENGTH = 3
+
+export type FullTextSearchProps = {
+  helpMessage?: string
+  helpUrl?: string
+}
+
+export const FullTextSearch: React.FunctionComponent<FullTextSearchProps> = ({
+  helpMessage = 'This search bar is powered by MySQL Full Text Search.',
+  helpUrl,
+}: FullTextSearchProps) => {
+  const { executeQueryRequest, queryMetadataQueryOptions } = useQueryContext()
+  const { showSearchBar } = useQueryVisualizationContext()
+  const [searchText, setSearchText] = useState('')
+  const searchInputRef = useRef<HTMLInputElement>(null)
+
+  const { data } = useQuery(queryMetadataQueryOptions)
+  const columnModels = data?.columnModels
+
+  const search = (event: React.SyntheticEvent<HTMLFormElement>) => {
+    event.preventDefault()
+
+    // The HTML validation doesn't trim the string, so we add an extra check.
+    // We don't auto-trim the form text ourselves because the user may still be focused on the input.
+    if (searchText.trim().length < MIN_SEARCH_QUERY_LENGTH) {
+      searchInputRef.current?.setCustomValidity(
+        `Search term must have a minimum of ${MIN_SEARCH_QUERY_LENGTH} characters`,
+      )
+    } else {
+      executeQueryRequest(queryBundleRequest =>
+        updateQueryUsingSearchTerm(
+          queryBundleRequest,
+          columnModels,
+          searchText,
+          setSearchText,
+        ),
```

Comment:
extracted logic into method `updateQueryUsingSearchTerm`. Logic in this method was not changed.